### PR TITLE
Zola Responsive Menu (iPhone compliant)

### PR DESCRIPTION
**Pull request by thibaultamartin**

### CODE REVIEW

File: templates/includes/header.html [hunk 1]
```diff
@@ -10,27 +10,12 @@
...
         {% set href = "/" ~ link.section ~ "/" %}
         {% set section = get_section(path=link.section ~ "/_index.md") %}
-        <details {{ util::current_open(if_path_starts_with=href) }} class="section-wrap mobile">
-            <summary class="section {{ util::current_class(if_path_starts_with=href) }}">
-                {{ link.title }}
+        <div class="section-wrap">
+            <input id="{{ link.slug }}-submenu-checkbox" type="checkbox" class="submenu-checkbox" aria-hidden="true">
+            <div class="submenu-title-container">
+                <label for="{{ link.slug }}-submenu-checkbox" class="submenu-title">{{ link.title }}</label>
                 <div class="arrow"></div>
```
Comment (jplatte) [inline templates/includes/header.html:17]: Why is the arrow not inside the `<label>`?

File: sass/_header.scss [hunk 2]
```diff
@@ -207,3 +187,50 @@
         }
     }
 }
+
+.submenu-checkbox {
+    display: none;
+
+    @media screen and (max-width: 1023px) {
+        &:not(:checked) + .submenu-title-container + .section-submenu-wrap {
```
Comment (jplatte) [inline sass/_header.scss:195]: Given you moved the different submenu checkboxes into distinct parent containers (`section-wrap`), you can shorten this to

```suggestion
        &:not(:checked) ~ .section-submenu-wrap {
```

File: sass/_header.scss [hunk 3]
```diff
@@ -207,3 +187,50 @@
...
+            display: none;
+        }
+
+        &:not(:checked) + .submenu-title-container {
+            .arrow {
+                background-image: url("/assets/down-arrow.svg");
+                width: 1rem;
+                height: 1rem;
+                margin: auto 0;
+                filter: invert(1);
```
Comment (jplatte) [inline sass/_header.scss:205]: Are we using the same image as black colored elsewhere? I'm almot inclined to recomment `{% include %}`ing it instead of using `background-image` and setting `fill: #fff` via CSS on the contained `<path>`. Also why does that file contain more than a dozen empty SVG groups? o_O

File: templates/includes/header.html (hunk 1 above)
Comment (thibaultamartin) [reply templates/includes/header.html:17]: Mostly to use `submenu-title-container` as a `flex` so I had more room to play with the arrow itself. Having a `div` inside a `label` also felt a little wrong.

Should I remove the container div and move the arrow inside the label?

File: sass/_header.scss (hunk 3 above)
Comment (thibaultamartin) [reply sass/_header.scss:205]: We are using the same image black coloured in the documentation yes. Is it more efficient to use `fill` than `background-image` and `invert`?

I'm going to give [scour](https://github.com/scour-project/scour) a go on these assets :)

Comment (thibaultamartin) [reply sass/_header.scss:205]:
```
Scour processed file "down-arrow.svg" in 12 ms: 602/857 bytes new/orig -> 70.2%
…
Scour processed file "up-arrow.svg" in 6 ms: 631/896 bytes new/orig -> 70.4%
```

File: templates/includes/header.html (hunk 1 above)
Comment (jplatte) [reply templates/includes/header.html:17]: I think you should yes, as right now tapping the arrow doesn't expand the menu.

File: templates/includes/header.html [hunk 4]
```diff
@@ -10,27 +10,10 @@
         {% if link.section %}
         {% set href = "/" ~ link.section ~ "/" %}
         {% set section = get_section(path=link.section ~ "/_index.md") %}
-        <details {{ util::current_open(if_path_starts_with=href) }} class="section-wrap mobile">
```
Comment (jplatte) [inline templates/includes/header.html:13]: You removed the use of `current_open`, maybe you should turn it into `current_checked` instead?

File: sass/_header.scss (hunk 3 above)
Comment (thibaultamartin) [reply sass/_header.scss:205]: I'm not sure I parsed the last sentence correctly, I read it as "you're using the wrong tool for the job, it's not a problem" :D

File: templates/includes/header.html (hunk 4 above)
Comment (thibaultamartin) [reply templates/includes/header.html:13]: This one actually always returned false, because `current_path` is not defined in `header.html`

File: sass/_header.scss (hunk 3 above)
Comment (jplatte) [reply sass/_header.scss:205]: Yes, that is what I was saying. You are using a much more powerful tool than necessary to do the coloring, so there is more potential for issues, but in practice it will likely be perfectly fine.

**Comment by thibaultamartin** [comment]
@jplatte whenever you have time, I could use a hand to make the mobile menu scrollable vertically before merging this one :)

**Review by jplatte — APPROVED** [review]
